spec/features/course/homepage_spec.rb

```diff
@@ -6,10 +6,23 @@
 
   with_tenant(:instance) do
     let(:course) { create(:course, :opened) }
-    let(:achievement_feed_notification) do
-      achievement = create(:course_achievement)
+    let(:feed_notifications) do
```

Comment:
Actually current one to me is optimised already, activity and notification are different factories, both of them have their own set of traits.

Factory coupling is one issue, the other is that there are N types of activities and N types of notifications, if we combine them into a new factory there will be N*N types.